src/czbenchmarks/tasks/task.py

```diff
@@ -21,6 +29,294 @@ class TaskOutput(BaseModel):
+                        type=type_info,
+                        default=field_info.default
+                        if field_info.default is not PydanticUndefined
+                        else None,
+                        required=field_info.is_required(),
+                    )
+
+            # 2. Get Baseline Parameters from the compute_baseline method signature
+            baseline_params = {}
+            try:
+                sig = inspect.signature(task_class.compute_baseline)
+                for param in list(sig.parameters.values())[1:]:
+                    if param.name in {
+                        "kwargs",
+                        "cell_representation",
+                        "expression_data",
+                    }:
+                        continue
+                    type_info = self._extract_type_info(param.annotation, param.name)
+                    baseline_params[param.name] = TaskParameter(
+                        type=type_info,
+                        default=param.default
+                        if param.default != inspect.Parameter.empty
+                        else None,
+                        required=param.default == inspect.Parameter.empty,
+                    )
+            except Exception as e:
+                # If baseline introspection fails, continue without baseline params
+                print(
+                    f"Warning: Could not introspect baseline parameters for {task_class.__name__}: {e}"
+                )
+
+            # 3. Get declared metric information from the static method
+            metrics = []
+            try:
+                metrics = [m.value for m in task_class.get_metric_types()]
+            except Exception as e:
+                print(
+                    f"Warning: Could not get metric types for {task_class.__name__}: {e}"
+                )
+
+            # 4. Get additional task metadata
+            description = self._extract_description(task_class)
+            display_name = getattr(task_class, "display_name", task_class.__name__)
+
+            return TaskInfo(
+                name=task_class.__name__,
+                display_name=display_name,
+                description=description,
+                task_params=task_params,
+                baseline_params=baseline_params,
+                metrics=metrics,
+            )
+        except Exception as e:
+            # Fallback task info if introspection fails
+            print(f"Warning: Task introspection failed for {task_class.__name__}: {e}")
+            return TaskInfo(
+                name=task_class.__name__,
+                display_name=getattr(task_class, "display_name", task_class.__name__),
+                description="Task introspection failed - please check task implementation",
+                task_params={},
+                baseline_params={},
+                metrics=[],
+            )
+
+    def _extract_type_info(self, annotation: Any, param_name: str) -> str:
```

Comment:
As I continue to read, I still think storing the type information as a `type` would be easier. I don't think you'd need this at all anymore